docs/migration/README.md

```diff
@@ -0,0 +1,830 @@
+```yaml
+humio-core:
+  authenticationMethod: oauth
+  oauthConfig:
+    autoCreateUserOnSuccessfulLogin: true
+    publicUrl: https://my-cluster.example.com
+  env:
+    - name: GOOGLE_OAUTH_CLIENT_SECRET
+      valueFrom:
+        secretKeyRef:
+          name: humio-google-oauth-secret
+          key: supersecretkey
+    - name: GOOGLE_OAUTH_CLIENT_ID
+      value: YOURCLIENTID
+```
+
+#### HumioCluster
+```yaml
+spec:
+  environmentVariables:
+    - name: AUTHENTICATION_METHOD
+      value: oauth
+    - name: AUTO_CREATE_USER_ON_SUCCESSFUL_LOGIN
+      value: "true"
+    - name: PUBLIC_URL
+      value: https://my-cluster.example.com
+    - name: GOOGLE_OAUTH_CLIENT_SECRET
+      valueFrom:
+        secretKeyRef:
+          name: humio-google-oauth-secret
+          key: supersecretkey
+    - name: GOOGLE_OAUTH_CLIENT_ID
+      value: YOURCLIENTID
+```
+
+### Authentication - OAuth Github
+
+#### Humio Helm Chart
+```yaml
+humio-core:
+  authenticationMethod: oauth
+  env:
+    - name: PUBLIC_URL
+      value: https://my-cluster.example.com
+    - name: GITHUB_OAUTH_CLIENT_ID
+      value: client-id-from-github-oauth
+    - name: GITHUB_OAUTH_CLIENT_SECRET
+      value: client-secret-from-github-oauth
+```
+
+#### HumioCluster
+```yaml
+spec:
+  environmentVariables:
+    - name: AUTHENTICATION_METHOD
+      value: oauth
+    - name: AUTO_CREATE_USER_ON_SUCCESSFUL_LOGIN
+      value: "true"
+    - name: PUBLIC_URL
+      value: https://my-cluster.example.com
+    - name: GITHUB_OAUTH_CLIENT_ID
+      value: client-id-from-github-oauth
+    - name: GITHUB_OAUTH_CLIENT_SECRET
+      value: client-secret-from-github-oauth
+```
+
+### Authentication - OAuth BitBucket
+
+#### Humio Helm Chart
+```yaml
+humio-core:
+  authenticationMethod: oauth
+  env:
+    - name: PUBLIC_URL
+      value: https://my-cluster.example.com
+    - name: BITBUCKET_OAUTH_CLIENT_ID
+      value: client-id-from-bitbucket-oauth
+    - name: BITBUCKET_OAUTH_CLIENT_SECRET
+      value: client-secret-from-bitbucket-oauth
+```
+
+#### HumioCluster
+```yaml
+spec:
+  environmentVariables:
+    - name: AUTHENTICATION_METHOD
+      value: oauth
+    - name: AUTO_CREATE_USER_ON_SUCCESSFUL_LOGIN
+      value: "true"
+    - name: BITBUCKET_OAUTH_CLIENT_ID
+      value: client-id-from-bitbucket-oauth
+    - name: BITBUCKET_OAUTH_CLIENT_SECRET
+      value: client-secret-from-bitbucket-oauth
+```
+
+### Authentication - SAML
+
+#### Humio Helm Chart
+```yaml
+humio-core:
+  authenticationMethod: saml
+  samlConfig:
+    publicUrl: https://my-cluster.example.com
+    idpSignOnUrl: https://accounts.google.com/o/saml2/idp?idpid=idptoken
+    idpEntityId: https://accounts.google.com/o/saml2/idp?idpid=idptoken
+  env:
+    - name: GOOGLE_OAUTH_CLIENT_SECRET
+      valueFrom:
+        secretKeyRef:
+          name: humio-google-oauth-secret
+          key: supersecretkey
+    - name: GOOGLE_OAUTH_CLIENT_ID
+      value: YOURCLIENTID
+```
+
+#### HumioCluster
+```yaml
+spec:
```

Comment:
I think we're missing the part about specifying the `IdpCertificateSecretName` on the HumioCluster? From what I can see the old helm chart assumed the secret holding the IDP certificate was called `idp-certificate` (see https://github.com/humio/humio-helm-charts/blob/69556adbcc2cc24bab4ce51c2a307653526fcb3d/charts/humio-core/templates/statefulset.yaml#L297-L301), whereas the operator by default assumes the secret is prefixed with the HumioCluster name which would be something like `examplecluster-idp-certificate` (see https://github.com/humio/humio-operator/blob/f087180c556d396004542e87850c52808f504345/pkg/controller/humiocluster/defaults.go#L167-L172).

Reply:
I think this should work by default since we call that same method when creating the pod: https://github.com/humio/humio-operator/blob/f087180c556d396004542e87850c52808f504345/pkg/controller/humiocluster/pods.go#L350

Reply:
Right, but we what I am missing here is instructions that tells the user to create the secret with the name matching what that function returns by default. The name for the secret in the old helm chart is not the same as what we assume the name is for the HumioCluster's.

Reply:
The old helm charts mounted in the k8s secret named `idp-certificate` whereas the operator assumes that secret is `fmt.Sprintf("%s-%s", hc.Name, idpCertificateSecretNameSuffix)`

Reply:
Also, `SAML_IDP_CERTIFICATE` is automatically appended to the env vars when saml is used.

Reply:
Ah got it. Will fix